File: WhiteLabelDemoApp/WhiteLabelDemoApp/LinkingCustomViewController.m
//
//  LinkingCustomViewController.m
//  WhiteLabelDemoApp
//
//  Created by ani on 7/12/16.
//  Copyright © 2016 LaunchKey. All rights reserved.
//

#import "LinkingCustomViewController.h"

@interface LinkingCustomViewController ()

@end

@implementation LinkingCustomViewController

@synthesize tfLinkingCode, tfDeviceName, switchDeviceName, btnLink;

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    //Navigation Bar Buttons
    UIBarButtonItem *leftItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"NavBack"] style:UIBarButtonItemStyleBordered target:self action:@selector(back:)];
    
    leftItem.tintColor = [[WhiteLabelConfigurator sharedConfig] getPrimaryTextAndIconsColor];
    
    [[self navigationItem] setLeftBarButtonItem:leftItem];
    
    //Navigation Bar Title
    UILabel* lbNavTitle = [[UILabel alloc] initWithFrame:CGRectMake(0,40,320,40)];
    lbNavTitle.textAlignment = NSTextAlignmentLeft;
    lbNavTitle.text = @"Linking View";
    lbNavTitle.textColor = [UIColor whiteColor];
    [lbNavTitle setFont:[UIFont boldSystemFontOfSize:18.0f]];
    self.navigationItem.titleView = lbNavTitle;
    self.navigationController.navigationBar.barTintColor = [[WhiteLabelConfigurator sharedConfig] getPrimaryColor];
    
    [switchDeviceName addTarget:self
                         action:@selector(stateChanged:) forControlEvents:UIControlEventValueChanged];
    [switchDeviceName setOnTintColor:[[WhiteLabelConfigurator sharedConfig] getSecondaryColor]];
    
    [btnLink setTitleColor:[[WhiteLabelConfigurator sharedConfig] getSecondaryColor]forState:UIControlStateNormal];
}

#pragma mark - Menu Methods
-(void)back:(id)sender
{
    [self.navigationController popViewControllerAnimated:YES];
}

#pragma mark - UISwitchDelegateMethods
- (void)stateChanged:(UISwitch *)switchState
{
    if ([switchState isOn])
    {
        tfDeviceName.enabled = true;
    }
    else
    {
        tfDeviceName.enabled = false;
        [tfDeviceName resignFirstResponder];
    }
}

#pragma mark - Button Methods
- (IBAction)btnLinkPressed:(id)sender
{
    NSString *qrCode = tfLinkingCode.text;
    
    if([qrCode length] == 7)
    {
        if ([switchDeviceName isOn])
        {
            NSString *deviceName = tfDeviceName.text;
            
            if([deviceName length] < 3)
            {
                UIAlertView *alert = [[UIAlertView alloc] initWithTitle:[NSString stringWithFormat:@"Device name should be at least 3 characters"]
                                                                message:nil
                                                               delegate:self
                                                      cancelButtonTitle:@"OK"
                                                      otherButtonTitles:nil];
                
                [alert show];
            }
            else if([deviceName length] == 0)
            {
                UIAlertView *alert = [[UIAlertView alloc] initWithTitle:[NSString stringWithFormat:@"Please enter a device name"]
                                                                message:nil
                                                               delegate:self
                                                      cancelButtonTitle:@"OK"
                                                      otherButtonTitles:nil];
                
                [alert show];
            }
            else
            {
                [[WhiteLabelManager sharedClient] linkUser:qrCode withDeviceName:deviceName withCompletion:^(NSError *error)
                 {
                     if(error != nil)
                     {
                         NSLog(@"error: %@", error);
                     }
                     else
                     {
                         [self back:self];
                     }
                 }];
            }
        }
        else
        {
            [[WhiteLabelManager sharedClient] linkUser:qrCode withDeviceName:nil withCompletion:^(NSError *error)
             {
                 if(error != nil)
                 {
                     NSLog(@"error: %@", error);
                 }
                 else
                 {
                     [self back:self];
                 }
             }];
        }
    }
    else
    {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:[NSString stringWithFormat:@"QR Code should be 7 characters"]
                                                        message:nil
                                                       delegate:self
                                              cancelButtonTitle:@"OK"
                                              otherButtonTitles:nil];
        
        [alert show];
    }
}


@end

File: WhiteLabelDemoAppSwift/SwiftWhiteLabelSDK/LinkingCustomViewController.swift
//
//  LinkingCustomViewController.swift
//  WhiteLabelDemoAppSwift
//
//  Created by ani on 7/31/16.
//  Copyright © 2016 LaunchKey. All rights reserved.
//

import Foundation

class LinkingCustomViewController: UIViewController
{
    @IBOutlet weak var tfLinkingCode: UITextField!
    @IBOutlet weak var tfDeviceName: UITextField!
    @IBOutlet weak var switchDeviceName: UISwitch!
    @IBOutlet weak var btnLink: UIButton!
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        self.title = "Linking View"
        
        //Navigation Bar Buttons
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "NavBack"), style: UIBarButtonItemStyle.Plain, target: self, action: #selector(ContainerViewController.back))
        self.navigationItem.leftBarButtonItem?.tintColor = WhiteLabelConfigurator.sharedConfig().getPrimaryTextAndIconsColor()
        
        switchDeviceName.addTarget(self, action: #selector(LinkingCustomViewController.stateChanged(_:)), forControlEvents: UIControlEvents.ValueChanged)
        switchDeviceName.onTintColor = WhiteLabelConfigurator.sharedConfig().getSecondaryColor()
        
        btnLink.setTitleColor(WhiteLabelConfigurator.sharedConfig().getSecondaryColor(), forState: .Normal)
        
    }
    
    override func didReceiveMemoryWarning()
    {
        super.didReceiveMemoryWarning()
    }
    
    
    @IBAction func btnLinkPressed(sender: AnyObject)
    {
        
        let qrCode = tfLinkingCode.text
        
        if(qrCode?.characters.count == 7)
        {
            if(switchDeviceName.on)
            {
                let deviceName = tfDeviceName.text
                
                if(deviceName?.characters.count < 3)
                {
                    let alert = UIAlertView()
                    alert.title = "Device name should be at least 3 characters"
                    alert.addButtonWithTitle("OK")
                    alert.show()
                }
                else if(deviceName?.characters.count == 0)
                {
                    let alert = UIAlertView()
                    alert.title = "Please enter a device name"
                    alert.addButtonWithTitle("OK")
                    alert.show()
                }
                else
                {
                    WhiteLabelManager.sharedClient().linkUser(qrCode, withDeviceName: deviceName, withCompletion: { (error) in
                        if((error) != nil)
                        {
                            print("\(error)")
                        }
                        else
                        {
                            self.back()
                        }
                    })
                }
            }
            else
            {
                WhiteLabelManager.sharedClient().linkUser(qrCode, withDeviceName:nil, withCompletion: { (error) in
                    if((error) != nil)
                    {
                        print("\(error)")
                    }
                    else
                    {
                        self.back()
                    }
                })
            }
        }
        else
        {
            let alert = UIAlertView()
            alert.title = "QR Code should be 7 characters"
            alert.addButtonWithTitle("OK")
            alert.show()
        }
    }
    
    func stateChanged(switchState: UISwitch)
    {
        if switchState.on
        {
            tfDeviceName.enabled = true
        }
        else
        {
            tfDeviceName.enabled = false
            tfDeviceName.resignFirstResponder()
        }
    }
    
    @IBAction func back()
    {
        if let navController = self.navigationController
        {
            navController.popViewControllerAnimated(true)
        }
    }
}

